### DISCUSSION TITLE
1. How to reconnect SignalR client after disconnect as fast as possible? 2. There are triggers for start of the session?

### DISCUSSION THREAD

**Discussion by supremeremixx** [theOehrly/Fast-F1#626] [Q&A] [answered]
Hello!
1. In important notes I read that after 2 hours SignalR client will be automatically disconnected by the server.
Can you please tell me how to reconnect to the server after disconnection as soon as possible? Maybe there are some auto reconnection function or flags which shows current connection state? Specific examples and code would be very much appreciated.

2. Also I would like to minimize the amount of connection time to the live timing. To do this, I need to start the SignalR client only after a new session has started. There are any triggers/flags that would signal the start of a new session?
For now I found method only with SignalR client - I trying to check "SessionStatus" every 10-15 mins. If previous status was "Ends" and after some time it changes to "Inactive" that means a creation of new session.

**Comment by theOehrly** [comment]
I don't really have a perfect solution for either problem.

1. When I needed to record for more than 2 hours, I just started multiple overlapping recordings. The data loading part is able to sort out the overlapping data so there is no problem here. I don't know of a way to reconnect without any data loss.
2. I simply manually schedule the start of the recording to something like 3-5 minutes before the start of the session as indicated by the event schedule. There is no automation or fancy logic that I use.

**Reply by supremeremixx** [reply]
1. Could you help with SignalR client settings for such usage please?
As I understood from documentation next set is needed:
- debug = False ("Only actual data in messages" means that it will storage only differences from previous message? For example if you try to connect to live timing right now, it will return 1 message with the last recorded data of the Belgium GP Race. If debug is set to True, this message is written to a txt file, but if debug = False, nothing is written because the data is no longer relevant and the first recorded message will only appear during FP1 of Dutch GP?)
- timeout = 0 (Disable auto disconnect when no new data available)
- outputfile = (same for every instance)
- another settings set by default.

So if I will create 1 instance of client every 1:50 and always trying to save the data in one file, clients will automatically detect overlapping data and save only actual data once right?

2. Got it. I'm trying to maximize automation that's why I interested in any flags which can indicates the real start of the session, because some of them may be canceled or rescheduled.

**Reply by theOehrly** [reply, marked as answer]
> So if I will create 1 instance of client every 1:50 and always trying to save the data in one file, clients will automatically detect overlapping data and save only actual data once right?

No, you would have each "instance" record data to a new file, not the same file.
The overlap is recognized during data loading. The ``LiveTimingData`` object can be provided with multiple files as input, as long as those are in chronological order.
See:
https://docs.fastf1.dev/livetiming.html#usage-example
https://docs.fastf1.dev/livetiming.html#fastf1.livetiming.data.LiveTimingData

I'm not aware of any flags that indicate the actual start of a session. But I also haven't looked for them. They may well exist. I just haven't really looked into automated live operation.

**Reply by theOehrly** [reply]
Ohh and the message recorded when ``debug=True`` is not from the last session. The recorder can't differentiate between current and past relevant or irrelevant messages. What you see recorded is just connection status data that contains no information about the actual data that is transported during an active session.

**Reply by supremeremixx** [reply]
Thanks a lot! I will record few files and use it with LiveTimingData.
